crates/dbsp/src/trace/layers/column_layer/tests.rs

```diff
@@ -9,39 +9,62 @@ use crate::{
         },
         Consumer, ValueConsumer,
     },
+    DBData, DBWeight,
 };
-use std::{cell::Cell, cmp::Ordering, ops::AddAssign, rc::Rc};
+use rkyv::{with::Skip, Archive, Deserialize, Serialize};
+use size_of::SizeOf;
+use std::ops::Add;
+use std::sync::{Arc, Mutex};
+use std::{cmp::Ordering, hash::Hash, ops::AddAssign};
 
 const TOTAL_TUPLES: usize = 100;
 const EXPECTED_DROPS: usize = TOTAL_TUPLES * 2;
 
-#[derive(Clone)]
+#[derive(Clone, Debug, Default)]
 struct Canary {
     /// The total number of drops done within the entire system
-    total: Rc<Cell<usize>>,
+    total: Arc<Mutex<usize>>,
 }
 
 impl Canary {
     fn new() -> Self {
         Self {
-            total: Rc::new(Cell::new(0)),
+            total: Arc::new(Mutex::new(0)),
         }
     }
 }
 
 impl Drop for Canary {
     fn drop(&mut self) {
-        self.total.set(self.total.get() + 1);
+        let mut ttl = self.total.lock().unwrap();
+        *ttl += 1;
     }
 }
 
-#[derive(Clone)]
-struct Item<T> {
+#[derive(Clone, Debug, Archive, Serialize, Deserialize, SizeOf)]
+struct Item<T>
+where
+    T: DBData + DBWeight,
```

Comment:
`DBDatat + DBWeight` here and in other places should be redundant since `DBWeight: DBData`